packages/query-core/src/queryObserver.ts

```diff
@@ -399,9 +400,21 @@ export class QueryObserver<
       !isValidTimeout(this.#currentRefetchInterval) ||
       this.#currentRefetchInterval === 0
     ) {
+      this.#clearRefetchInterval()
       return
     }
 
+    // Don't bother clearing the interval and re-setting it if
+    // we already have a properly set interval running
+    if (
+      prevInterval == this.#currentRefetchInterval &&
+      this.#refetchIntervalId
+    ) {
+      return
+    }
+
+    this.#clearRefetchInterval()
+
```

Comment:
I don’t think this is correct. The previous implementation re-sets an interval every time new data comes in, which this implementation does not. Suppose the following scenario:

- we have a refetchInterval set to every 5 minutes (timestamp = 0)
- after 3 minutes, we get another update into our cache (e.g. with setQueryData or with an invalidation after a mutation) (timestamp = 3 min)

--> that is supposed to reset the interval, so that we don’t get another refetch after 2 minutes (timestamp = 5min), but 5 minutes _after the last update_ (so timestamp = 8min)

I hope we have a test case for this that now fails, otherwise, we should probably add one :)